content/articles/how-to-add-support-for-spatial-data-in-your-database-using-geoalchemy-and-spatialite/index.md

```diff
@@ -0,0 +1,397 @@
+images:
+  - url: /engineering-education/how-to-build-a-clock-with-javascript-and-svg/hero.png
+    alt: Building a clock using JavaScript and SVG
+---
+
+When building certain kinds of applications, it is important to know what kind of data you'll be working with, so you can make adequate preparations to hold and process them. One such instance is when you need to work with "spatial data" - a typical example of which might be: storing location ( in terms of longitude and latitude values).
+<!--more-->
+In this article, we will use Sqlite and its spatial extension to create a database that supports spatial data read and write operations, for a simple flask server.
+
+
+### Table of Contents
+
+- [Introduction](#introduction)
+- [Building A Simple Web Server](#building-a-simple-web-server)
+- [Adding Support For Spatial Data](#adding-support-for-spatial-data)
+- [Conclusion](#conclusion)
+
+
+### Introduction
+
+When building certain kinds of applications, it is important to know what kind of data you'll be working with, so you can make adequate preparations to hold and process them. One such instance is when you need to work with "spatial data" - a typical example of which might be: storing location ( in terms of longitude and latitude values). In addition, you may want to be able to make queries based on these kinds of information, for example, you could imagine querying your database for "10 closest entities to a given location (lon,lat)".
+
+How does one make provisions for such data?, the aim of this article is to show you how to implement a spatial database in your development environment, with the aid of a typical Flask server using GeoAlchemy and Spatialite (for development purposes using SQLite).
+
+
+### Building A Simple Web Server
+We'll start by creating a simple backend server for a service-hailing platform, where users can connect with 'service providers' in close proximity to them, and get them to offer their services to them for a fee. For this kind of platform, we have to deal with spatial information like "location of client", "location of service-providers", and we might also want to make queries to the database to perhaps find "the closest 10 service-providers" to a given location for example. We can only achieve this if we set things up correctly and add support for these kind of information to our database, since they don't directly have that kind of support for spatial data.
+
+#### Pre-Requisites
+- Python3 Installed
+- Basic Flask Knowledge
+
+We'll start by creating a folder called `server` and in this folder, we'll add a new folder called `core` and three other files called `wsgi.py`, `models.py` and `config.py`. Inside the `core` folder, add two new files called: `__init__.py`, and `views.py`. Once you're done your `server` folder should have a structure like this one below:
+
+```bash
+-server/
+    -core/
+        -__init__.py
+        -views.py
+    -models.py
+    -wsgi.py
+    -config.py
```

Comment:
You can use `|` and `-` for file structure representations. Search it in other Section articles or just Google it